### PR TITLE
GUVNOR-2935: Error when loading stock examples on Windows

### PR DESCRIPTION
**Pull request by paulovmr**

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImpl.java [hunk 1]
```diff
@@ -203,190 +190,194 @@ public ExampleRepository getPlaygroundRepository() {
...
 
-    private String getExampleAlias( final String repositoryURL ) {
+    String getExampleAlias(final String repositoryURL) {
         String alias = repositoryURL;
-        alias = alias.substring( alias.lastIndexOf( '/' ) + 1 );
-        final int lastDotIndex = alias.lastIndexOf( '.' );
-        if ( lastDotIndex > 0 ) {
-            alias = alias.substring( 0,
-                                     lastDotIndex );
+        alias = alias.substring(alias.lastIndexOf(getFileSeparator()) + 1);
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImpl.java]: Real change is here.

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImpl.java [hunk 2]
```diff
@@ -203,190 +190,194 @@ public ExampleRepository getPlaygroundRepository() {
...
+        if (lastDotIndex > 0) {
+            alias = alias.substring(0,
+                    lastDotIndex);
         }
         return "examples-" + alias;
     }
 
-    private Set<ExampleProject> convert( final Set<Project> projects ) {
+    String getFileSeparator() {
+        return FileSystems.getDefault().getSeparator();
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImpl.java]: Real change is here.

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImplTest.java [hunk 3]
```diff
@@ -102,370 +92,390 @@ public void fire( final NewProjectEvent event ) {
...
+                any(RepositoryEnvironmentConfigurations.class));
+        verify(ioService,
+                times(1)).startBatch(any(FileSystem.class));
+        verify(ioService,
+                times(1)).endBatch();
+        verify(newProjectEvent,
+                times(2)).fire(any(NewProjectEvent.class));
     }
 
+    @Test
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImplTest.java]: New test.

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImplTest.java [hunk 4]
```diff
@@ -102,370 +92,390 @@ public void fire( final NewProjectEvent event ) {
...
+        doReturn("\\").when(service).getFileSeparator();
+
+        final String repositoryURL = "C:\\folder\\repo.git";
+
+        final String alias = service.getExampleAlias(repositoryURL);
+
+        assertEquals("examples-repo", alias);
+    }
+
+    @Test
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImplTest.java]: New test.

**Comment by tomasdavidorg** [comment]
Hi @paulovmr, does it fix this [GUVNOR-2921](https://issues.jboss.org/browse/GUVNOR-2921)? Is it the same issue?

I am not sure with the code style. It looks like both files don't use the kie codestyle.

Thanks.

**Comment by paulovmr** [comment]
Hi @tomasdavidorg , yes, it's the same issue, didn't see that there was already an open issue for this. And yes, both were formatted using the new kie codestyle.

**Comment by paulovmr** [comment]
@tomasdavidorg Actually I used the wrong formatter... I will fix this and the new conflicts.

**Comment by paulovmr** [comment]
jenkins retest this

**Pull request merged**